examples/api_test_sm/test_smapi.cu

```diff
@@ -0,0 +1,114 @@
+/*
+ * @brief Subgraph matching test for shared library advanced interface
+ * @file test_smapi.cu
+ */
+
+#include <stdio.h>
+
+#include <thrust/copy.h>
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h>
+
+#include <gunrock/gunrock.h>
+#include <gunrock/app/sm/sm_app.cu>
+#include <gunrock/app/test_base.cuh>
+
+// __global__ 
+// void kernel_print(unsigned long* list_subgraphs, unsigned long* size) {
+//   int tid = threadIdx.x;
+//   if(tid > size[0]) return;
+//   printf("%lu\n", list_subgraphs[tid]);
+// }
+
+int main(int argc, char *argv[]) {
+  int num_data_nodes = 5, num_data_edges = 10;
+  int data_row_offsets[6] = {0, 2, 6, 7, 9, 10};
+  int data_col_indices[10] = {1, 3, 0, 2, 3, 4, 1, 0, 1, 1};
+
+
+
+  // auto d_row_offsets_ptr = device_data_row_offsets.data().get();  // thrust::device_ptr()... a literal raw pointer (c++) int*, 
+  //int* d_row_offsets_ptr = device_data_row_offsets.data().get();  // thrust::raw_pointer_cast();
+
+  //cudaMalloc version, not used
+  // int* device_data_row_offsets;
+  // cudaMalloc((void**) &device_data_row_offsets, sizeof(int)*6);
+  // to, from, size, type
+  // cudaMemcpy(device_data_row_offsets, data_row_offsets, sizeof(int)*6, cudaMemcpyHostToDevice);
+
+  //printing contents of a thrust array that's currently allocated on device
+  //thrust::copy(device_data_row_offsets.begin(), device_data_row_offsets.end(), std::ostream_iterator<int>(std::cout, " "));
+
+  int num_query_nodes = 3, num_query_edges = 6;
+  int query_row_offsets[4] = {0, 2, 4, 6};
+  int query_col_indices[6] = {1, 2, 0, 2, 0, 1};
+
+
+
+  unsigned int location;
+  //unsigned int device = 0x02;  // CPU = 1, GPU = 2. Rename location.
+
+  if(*argv[1] == '1'){
+    //allocating arrays on CPU. If location ==2, need to allocate on gpu
+    location = 0x01;
+    
+    unsigned long *sm_counts = (unsigned long *)malloc(sizeof(unsigned long));
+    unsigned long *list_sm = (unsigned long *)malloc(sizeof(unsigned long)); 
+    
+    double elapsed =
+      sm(num_data_nodes, num_data_edges, data_row_offsets, data_col_indices,
+          num_query_nodes, num_query_edges, query_row_offsets, query_col_indices,
+          1, sm_counts, list_sm, location);
+    
+    // why access 0th element instead of dereferencing the pointer?
+    printf("Number matched subgraphs: [%d]\n", sm_counts[0]);
+    printf("List of matched subgraphs: [%d]\n", list_sm[0]);
+  }
+
+  if(*argv[1] == '2'){
+
+    location = 0x02;
```

Comment:
My understanding was that this isn't functional? Can you link to the issue in a comment elaborating that.

Reply:
Definitely - I couldn't find a way to make the sm app run with the device flag [(link to issue)](https://github.com/gunrock/gunrock/issues/806). I can remove the sm test from the pr since it doesn't work with sm currently.